Kernel/LPO.cpp

```diff
@@ -79,35 +79,44 @@ Ordering::Result LPO::comparePrecedences(Term* t1, Term* t2) const
 } // LPO::comparePrecedences
 
 Ordering::Result LPO::compare(TermList tl1, TermList tl2) const
+{
+  return compare(AppliedTerm(tl1),AppliedTerm(tl2));
```

Comment:
This might not be completely zero-cost as you'll have to do some null checks in the `AppliedTerm` version. But it's probably fine - if it shows up as a hit in profiling we can probably do some template magic to smooth it out.

Reply:
Added a separate constructor for this case, hopefully it solves it.